modules/core/arrow-core-data/src/main/kotlin/arrow/typeclasses/MonadLogic.kt

```diff
@@ -0,0 +1,188 @@
+package arrow.typeclasses
+
+import arrow.Kind
+import arrow.core.ForOption
+import arrow.core.Option
+import arrow.core.Tuple2
+import arrow.core.fix
+
+/**
+ * MonadLog is a typeclass that extends a MonadPlus. It provides functions to control
+ * when computations should be performed.
```

Comment:
This is a bit weak imo. A better description would be that it improves upon `MonadPlus` with fairness guarantees.